.github/workflows/common_merge.yaml

```diff
@@ -0,0 +1,120 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Common Merge Workflow
+# Collects common Merge related jobs for the NWX stack, specifically
+# 1. Bumping version tags
+# 2. Deploying updated documentation
+
+on:
+  workflow_call:
+    inputs:
+      doc_target:
+        description: "The name of the documentation target. Set to 'Sphinx' to skip doxygen"
+        type: string
+        required: false
+        default: 'Sphinx'
+      generate_module_docs:
+        description: "Whether or not to generate PluginPlay Module documentation"
+        type: boolean
+        default: false
+    secrets:
+      CMAIZE_GITHUB_TOKEN:
+        description: "Token passed to CMaize"
+        required: false
+      TAG_TOKEN:
+        description: "Token used to bump the version tag"
+        required: true
+      CONTAINER_REPO_TOKEN:
+        description: "Token to access Github Image Registry"
+        required: true
+
+jobs:
+  # Bump the version tag
+  bump_tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: haya14busa/action-bumpr@v1
+        with:
+          default_bump_level: patch
+          github_token: ${{ secrets.TAG_TOKEN }}
+          tag_as_user: .github[bot]
+          tag_as_email: .github[bot]@github.com
+
+  # Deploy updated documentation
+  deploy_docs:
+    runs-on: ubuntu-latest
+    container: 
+      image: ghcr.io/nwchemex-project/nwx_buildenv:latest
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.CONTAINER_REPO_TOKEN }}
+    steps:
+      - uses: actions/checkout@v4
+      # These next two steps will configure CMake if required
+      - name: Build Only Docs
+        if: |
+          inputs.generate_module_docs == 'false' && inputs.doc_target != 'Sphinx'
+        run: |
+          cmake -Bbuild -H. -GNinja -DBUILD_DOCS=ON -DONLY_BUILD_DOCS=ON
+        shell: bash
+      - name: Build Library
+        if: inputs.generate_module_docs == true
+        env:
+          CMAIZE_GITHUB_TOKEN: ${{secrets.CMAIZE_GITHUB_TOKEN}}
+        run: |
+          toolchain=/toolchains/nwx_gcc-11.cmake
+          echo 'set(CMAIZE_GITHUB_TOKEN '${CMAIZE_GITHUB_TOKEN}')' >> $toolchain
+
+          cmake -Bbuild -H. -GNinja -DBUILD_DOCS=ON \
+          -DCMAKE_INSTALL_PREFIX=./install \
+          -DCMAKE_TOOLCHAIN_FILE="${toolchain}"
+      # Build the Doxygen docs
+      - name: Build Doxygen Docs
+        if: ${{ inputs.doc_target != 'Sphinx' }}
+        run: |
+          cmake --build build --target ${{ inputs.doc_target }} --parallel
+
+          # Migrate the Doxygen documentation to the docs source
+          mkdir docs/build
+          mkdir docs/build/html
+          mv build/html "docs/build/html/${{ inputs.doc_target }}"
+        shell: bash
```

Comment:
I realize it's a fairly small amount of code duplication between this workflow and `common_pull_request.yaml`, but should the building of Doxygen and Sphinx be factored out?

Reply:
I don't really think it's work it, personally. It'd end being an action to wrap only a handful of line. Once you add the boiler plate for the `action.yaml`, I don't know how much it reduces.

Reply:
It's not necessarily about reducing the lines of code as much as it is about encapsulating the logic, i.e., making sure we don't have to remember to add new flags in two places. That said, I can go either way on this one. Aside from turning on warnings for undocumented things (and maybe making warnings into errors), I don't anticipate needing to muck with these commands too often.